### PR TITLE
deposit: Fetch curation request periodically and on record update

### LINKED ISSUE #26: Publish button does not change after saving draft
When the publish button is active, it means that the curation request for this record has been accepted.
If changes are made and the draft is saved again, the curation request gets resubmitted. Hence, the publish button should not be visible anymore but the view request element should be shown.

### Possible solution
Fetch the curation request for the record
- after saving the draft
- every few seconds (similar to the timeline on the request detail page). This allows to always be up-to-date without actively refreshing the page - even updating the deposit box automatically from showing the `view request` element to showing the publish button

### PR DESCRIPTION
**Pull request by rekt-hard**
closes #26

Fetch curation request every 10 seconds, to get updates without refreshing the page manually.
Fetch curation request after record state has changed on the deposit form. This usually happens when the draft is saved. It allows to get an instant update on the request and render the UI appropriately, instead of having to wait until the next periodic tick happens.

In the video, left side shows a user on the deposit view and right side shows the a curator on curation request view. A curator starts review and accepts the review, which will be displayed on the deposit fetch after the request is fetched again. Then on the deposit page, the record is updated and saved, which will fetch the request immediately. 

[Screencast from 2024-10-03 16:04:10.webm](https://github.com/user-attachments/assets/621586f9-449a-4a60-8867-651a5df25196)

### CODE REVIEW

File: invenio_curations/assets/semantic-ui/js/invenio_curations/deposit/DepositBox.js [hunk 1]
```diff
@@ -100,10 +120,23 @@ export class DepositBoxComponent extends React.Component {
...
+      this.resetInterval();
+      this.fetchCurationRequest();
+    }
+  };
+
   render() {
     const { latestRequest } = this.state;
     const { record, permissions, groupsEnabled } = this.props;
 
+    this.checkShouldFetchCurationRequest();
```
Comment (max-moser) [inline invenio_curations/assets/semantic-ui/js/invenio_curations/deposit/DepositBox.js:138]: does this trigger on keypress in the TinyMCE fields like description, or just on something like focus loss?
if it's on keypress/character input then that may be a bit much?

Comment (rekt-hard) [reply invenio_curations/assets/semantic-ui/js/invenio_curations/deposit/DepositBox.js:138]: Form updates are only triggered on specific events like selecting a community or saving a record. After these events, the props will be updated and the timestamp will be set. Then the request will be fetched. So we are good here :)

Comment (rekt-hard) [reply invenio_curations/assets/semantic-ui/js/invenio_curations/deposit/DepositBox.js:138]: A more appropriate solution would be to check the `record.updated` field, which is usually set on the response from the back end. However, this does not seem to get updated on the local record, as it is not de-serialized from the response.

Created a PR to address this: https://github.com/inveniosoftware/invenio-rdm-records/pull/1838
I would also prefer to use the `updated` field as it allows to get rid of the additional prop.

**Review by max-moser — APPROVED** [review]
thanks a lot for this, and the clarification about the frequency of updates!
if an API call had been sent on every keystroke, that might have been a bit much – but if it's just on every "relevant" event, then we're golden!

**Pull request merged**